src/core/taint.c

```diff
@@ -31,8 +32,8 @@ static int short_uid_gid_range(UIDRangeUsernsMode mode) {
         return !uid_range_covers(p, 0, 65535);
 }
 
-char** taint_strv(void) {
-        const char *stage[12] = {};
+char** taint_strv(const Manager *m) {
```

Comment:
No. "Taint" is supposed to be some OS-wide thing, about legacy/improper setups. Dependency cycles do not really belong here. Instead, why not add a flag for SystemState if _default transaction_ had a dep cycle?

Reply:
> "Taint" is supposed to be some OS-wide thing, about legacy/improper setups.

Exactly. A cycle is very much an "improper setup". Usually this occurs in the early boot or late shutdown and causes random jobs to be skipped. A taint seems suitable.